ethcore/light/src/net/mod.rs

```diff
@@ -766,15 +820,20 @@ impl LightProtocol {
 			awaiting_acknowledge: None,
 		}));
 
-		for handler in &self.handlers {
+		let any_kept = self.handlers.iter().map(
+			|handler|
 			handler.on_connect(&Ctx {
 				peer: *peer,
 				io: io,
 				proto: self,
 			}, &status, &capabilities)
-		}
+		).fold(PeerStatus::Kept, PeerStatus::bitor);
```

Comment:
not sure if this is intentional, but it does correctly handle the caveat where having no attached handlers should keep the peer (as light servers don't have any handlers but shouldn't disconnect incoming connections).